forem/forem
# Add new cache key for article sidebar & bottom_content

## What type of PR is this? (check all applicable)

## Description
This adds a new concept of busting caches for an author's cached content areas, specifically the article's sidebar "More from [author]" and bottom content "Read next" areas. The idea here is to rely more on busting the timestamp `user/organization.latest_article_updated_at`, and therefore can lead to less stale content (broken links, incorrect titles and tags, usernames, etc.). For now, this simply clears the cached areas when a title and tag list is updated. 

I realize this won't cover all cases, but I think this is a good start, and I've also been sitting on this code for too long. Would definitely appreciate feedback here!

## Related Tickets & Documents
Resolves https://github.com/forem/internalEngineering/issues/242

## QA Instructions, Screenshots, Recordings
1. Enable caching in development: `rails dev:cache`
2. Mock the articles that will be viewed in:
- `_sticky_nav`: return early in `get_user_stickies.rb, call` method (line 4) with `return author.articles.published.last(3)`
- `_bottom_content`: replace line 210 in `app/views/articles/show.html.erb` with `suggested_articles = (@organization || @user).articles.published.last(3)`
3. Create and publish a new article via `/new`
4. - [ ] See that it shows up in the sidebar and bottom content areas
5. Update the created article's title and tags
6. - [ ] See that the new changes are reflected in the sidebar and bottom content areas


### UI accessibility concerns?
Nope! This doesn't change any front end code.

## Added tests?
- [x] I need help with writing tests. Not sure how to write a test for caching.

## Added to documentation?

## [optional] Are there any post deployment tasks we need to perform?
We will have to monitor our alerting, and maybe turn it off for a bit. The last time we touched `/articles/_sticky_nav.html.erb` the alerting went off a lot, but the site didn't slow down much. cc @maestromac here.

## [optional] What gif best describes this PR or how it makes you feel?
![I'm unsure :(](https://media.giphy.com/media/Ykz1yddXHehPPEMlVb/giphy.gif)
